### ISSUE TITLE
Nice Work!

### ISSUE THREAD

**Issue by leongatys** [jcjohnson/neural-style#7]
Maybe try our normalised VGG network for better results: https://bethgelab.org/deepneuralart/

Best
Leon

**Comment by jlukic** [comment]
:+1:

**Comment by jcjohnson** [comment]
Thanks! Since the file size is much smaller, I assume that you removed the fully connected layers? That should also help the people who have been running into memory problems.

**Comment by alexjc** [comment]
Even with the old model, for memory problems, I did the following:
* Hack loadcaffe to not call `:cuda()` on the network as its loaded.
* Instead in `neural_style` only call `:cuda()` on layers used before `net:add(layer)`.

The new optimization algorithm, `lbfgs` consumes way more memory too, you could revert to SGD.

cc. @auduno @paperkettle

**Comment by jcjohnson** [comment]
@alexjc Those sound like good ways to reduce memory load.

SGD works, but I'm hesitant to add it back because the results are so much worse and the hyperparameters needed to get good results are quite different.

**Comment by alexjc** [comment]
Yeah, the results are much better, but only the smallest images fit in memory on 1GB GPUs and the parameters in the repo by default just don't work as well on small images. (Size dependent parameters.)

**Comment by jcjohnson** [comment]
@leongatys What kind of preprocessing does this network expect? I'm having a hard time getting better results than with the base VGG-19 network.

**Comment by leongatys** [comment]
In general there is nothing different to this network in terms of input-output mapping, but it might make finding good hyperparameters easier, because you know more about the size of the activations in each layer. 
Another important factor is at which scale you extract the style features, ie. how you downsample/resize the style image for a given content image.

**Comment by alexjc** [comment]
@leongatys What about standardizing inputs? VGG expected this to be done based on the mean pixel color.  Do you do any normalization of the matrix based on the number of units in the layers?

Thanks for taking the time to comment! It's fasinating work :-)

**Comment by jcjohnson** [comment]
@leongatys Thanks for the tip on style scaling - it's really easy to do and makes a huge difference! Implemented in https://github.com/jcjohnson/neural-style/commit/dba29fe3d6bc8f8a9fa5d1f9f2ddd0b070477d3d.

**Comment by kylemcdonald** [comment]
wow, very curious to see if the same results can be generated with the network that's had the fc layers stripped -- it's small enough to fit in gpu memory on my laptop! thanks @leongatys @jcjohnson for your work :)

**Comment by auduno** [comment]
sorry to hijack this thread, but from your papers @leongatys I get the impression you used Caffe. Curious whether you have a working implementation of L-BFGS in Caffe? If so it would be great to see it merged into Caffe, see e.g. https://github.com/BVLC/caffe/issues/1121

And awesome work both @jcjohnson and @leongatys, thanks for sharing!

**Comment by JulienBreux** [comment]
👍

**Comment by msurguy** [comment]
@leongatys is it true that a patent for this particular implementation is pending?

**Comment by Danderssen** [comment]
Was anybody able to achieve good results with this net? I've been trying quite some different parameter sets without success. Does anybody mind sharing their parameters?

Thanks!

**Comment by alexjc** [comment]
@Danderssen Not with this implementation, I use a Recipe for Lasagne that was built with it from scratch and results are good: https://github.com/Lasagne/Recipes

I don't really notice significant difference though.
